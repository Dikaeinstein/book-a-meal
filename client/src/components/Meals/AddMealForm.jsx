import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Form, Field } from 'formik';
import { ProgressBar } from 'react-materialize';
import Label from '../util/Label';
import Button from '../util/Button';
import Loading from '../util/Loading';
import cloudinaryImageUpload from '../../helpers/cloudinaryImageUpload';

export class AddMealForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      imageUploading: false,
      imageUploaded: false,
      imageUploadError: false,
      imageUrl: '',
    };
    this.handleImageChange = this.handleImageChange.bind(this);
  }

  async handleImageChange(event) {
    try {
      const file = event.target.files[0];

      this.setState({ imageUploading: true, imageUploaded: false });
      // upload image and get imageUrl
      const imageUrl = await cloudinaryImageUpload(file);
      this.props.values.imageUrl = imageUrl;
      this.setState({
        imageUploading: false,
        imageUploaded: true,
        imageUrl,
      });
    } catch (error) {
      this.setState({
        imageUploadError: true,
        imageUploading: false,
        imageUploaded: false,
      });
    }
  }

  render() {
    const {
      errors,
      touched,
      isSaving,
    } = this.props;

    return (
      <Form className="form card">
        <div>
          <Label className="label label-block" htmlFor="name">
            Meal name:
          </Label>
          <Field
            placeholder="Enter meal name"
            name="name"
            required
            className={touched.name
            && errors.name ? 'input-error' : ''}
          />
          {
            touched.name
            && errors.name
            && <div className="error">{errors.name}</div>
          }
        </div>
        <div>
          <Label className="label label-block" htmlFor="price">
          Price (&#x20a6;):
          </Label>
          <Field
            type="number"
            placeholder="Enter price"
            name="price"
            required
            className={touched.price
            && errors.price ? 'input-error' : ''}
          />
          {
            touched.price
            && errors.price
            && <div className="error">{errors.price}</div>
          }
        </div>
        <div>
          <Label className="label label-block" htmlFor="description">
            Description:
          </Label>
          <Field
            component="textarea"
            placeholder="Enter price"
            name="description"
            required
            className={touched.description
            && errors.description ? 'input-error' : ''}
          />
          {
            touched.description
            && errors.description
            && <div className="error">{errors.description}</div>
          }
        </div>
        <div>
          <Label className="label label-block" htmlFor="image">
            Image:
          </Label>
          <input
            type="file"
            name="image"
            accept="image/*"
            onChange={this.handleImageChange}
          />
          {
            this.state.imageUploading
            ?
              <div className="text-dark text-center uploading">
                <small>
                  <Loading text="uploading image . . .">
                    <ProgressBar />
                  </Loading>
                </small>
              </div>
            : null
          }
          {
            this.state.imageUploaded ?
              <div className="text-dark">
                <div className="text-center">
                  <small>Image uploaded</small>
                </div>
                <img
                  src={this.state.imageUrl}
                  alt="meal preview"
                  className="meal-preview center"
                />
              </div> : null
          }
          {
            this.state.imageUploadError ?
              <div className="text-danger">
                <div className="text-center">
                  <small className="error">
                    Error uploading image, please try again
                  </small>
                </div>
              </div> : null
          }
        </div>
        <Button
          value="Add meal"
          type="submit"
          className="btn btn-default font-weight-bold"
          data-test="addmeal-submit"
          disabled={isSaving}
        />
        {isSaving ?
          <Loading text="Saving Meal . . .">
            <ProgressBar />
          </Loading>
          : null}
        {!isSaving && errors.addMeal ?
          <div className="error text-center">
            <small>{errors.addMeal}</small>
          </div>
          : null}
      </Form>
    );
  }
}

AddMealForm.propTypes = {
  errors: PropTypes.objectOf(PropTypes.string).isRequired,
  touched: PropTypes.objectOf(PropTypes.bool).isRequired,
  isSaving: PropTypes.bool.isRequired,
  /* eslint react/forbid-prop-types: 0 */
  values: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  isSaving: state.meals.isSaving,
});

export default connect(mapStateToProps)(AddMealForm);
