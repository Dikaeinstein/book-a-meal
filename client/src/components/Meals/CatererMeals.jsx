import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Preloader } from 'react-materialize';
import { fetchMeals } from '../../actions/mealActions';
import CatererMealList from './CatererMealLIst';
import Loading from '../util/Loading';

class ConnectedCatererMeals extends Component {
  componentDidMount() {
    this.props.fetchMeals();
  }

  render() {
    const { isFetching, handleMealUpdate } = this.props;
    if (isFetching) {
      return (
        <div className="loader-container">
          <Loading text="Loading...">
            <Preloader flashing size="big" />
          </Loading>
        </div>
      );
    }

    return (
      <section className="bg-light landing-main">
        <h1 className="text-center">
          Meals
        </h1>
        <CatererMealList handleMealUpdate={handleMealUpdate} />
      </section>
    );
  }
}

ConnectedCatererMeals.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  fetchMeals: PropTypes.func.isRequired,
  handleMealUpdate: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  isFetching: state.meals.isFetching,
});

const mapDispatchToProps = dispatch => ({
  fetchMeals: () => dispatch(fetchMeals()),
});

const CatererMeals = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ConnectedCatererMeals);

export default CatererMeals;
