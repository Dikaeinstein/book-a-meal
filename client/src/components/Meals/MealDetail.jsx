import React from 'react';
import PropTypes from 'prop-types';

const MealDetail = ({ meal, closeMealDetail }) => {
  const handleCloseMealDetail = () => closeMealDetail();

  const mealDetailItemStyle = {
    padding: '0 1rem 1rem 1rem',
  };

  return (
    <div className="flex-container">
      <div style={{ ...mealDetailItemStyle, flex: '60%' }}>
        <img src={meal.imageUrl} alt={meal.name} />
      </div>
      <div style={{ ...mealDetailItemStyle, flex: '40%', color: '#000' }}>
        <h4
          className="text-capitalize font-weight-bold"
          style={{ margin: '.875rem 0' }}
        >
          {meal.name}
        </h4>
        <p
          className="font-weight-bold"
          style={{ margin: '.5rem 0' }}
        >
          Description:
        </p>
        <p style={{ fontSize: '.875rem', margin: '.5rem 0' }}>
          {meal.description}
        </p>
        <p style={{ margin: '.875rem 0' }}>
          <span className="font-weight-bold">
            Caterer:{' '}
          </span>
          <span>{meal.user.name}</span>
        </p>
        <div
          onClick={handleCloseMealDetail}
          onKeyPress={handleCloseMealDetail}
          className="text-dark font-weight-bold"
          style={{ cursor: 'pointer', fontSize: '.85rem' }}
          role="button"
          tabIndex="0"
        >
          Back to meals
        </div>
      </div>
    </div>
  );
};

MealDetail.propTypes = {
  meal: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.object,
  ])).isRequired,
  closeMealDetail: PropTypes.func.isRequired,
};

export default MealDetail;
