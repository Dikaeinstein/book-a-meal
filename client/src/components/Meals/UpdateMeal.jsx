import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Formik } from 'formik';
import UpdateMealForm from './UpdateMealForm';
import mealSchema from '../../validation/mealSchema';
import { updateMeal } from '../../actions/mealActions';


const ConnectedUpdateMeal = ({ editMeal, meal }) => {
  const handleSubmit = (values, actions) => {
    editMeal(values, actions, meal.id);
  };
  return (
    <section className="add-section">
      <h2 className="text-center">Update Meal</h2>
      <Formik
        initialValues={meal}
        validationSchema={mealSchema}
        component={UpdateMealForm}
        onSubmit={handleSubmit}
      />
    </section>
  );
};

ConnectedUpdateMeal.propTypes = {
  meal: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string, PropTypes.number])).isRequired,
  editMeal: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  editMeal: (values, actions, id) => dispatch(updateMeal(values, actions, id)),
});

const UpdateMeal = connect(null, mapDispatchToProps)(ConnectedUpdateMeal);

export default UpdateMeal;
