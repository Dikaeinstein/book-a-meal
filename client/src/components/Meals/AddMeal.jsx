import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Formik } from 'formik';
import swal from 'sweetalert';
import AddMealForm from './AddMealForm';
import mealSchema from '../../validation/mealSchema';
import { addMeal } from '../../actions/mealActions';


const ConnectedAddMeal = ({ saveMeal, closeModal }) => {
  const handleSubmit = async (values, actions) => {
    await saveMeal(values, actions);
    await swal({
      text: 'Meal added successfully',
      icon: 'success',
      className: 'swal-button--confirm',
    });
    return closeModal();
  };
  return (
    <section className="add-section">
      <h2 className="text-center">Add New Meal</h2>
      <Formik
        initialValues={{
          name: '',
          description: '',
          price: '',
          imageUrl: undefined,
        }}
        validationSchema={mealSchema}
        component={AddMealForm}
        onSubmit={handleSubmit}
      />
    </section>
  );
};

ConnectedAddMeal.propTypes = {
  saveMeal: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  saveMeal: (values, actions) => dispatch(addMeal(values, actions)),
});

const AddMeal = connect(null, mapDispatchToProps)(ConnectedAddMeal);

export default AddMeal;
