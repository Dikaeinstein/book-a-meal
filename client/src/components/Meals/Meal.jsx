import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import { connect } from 'react-redux';
import Button from '../util/Button';
import MealDetail from './MealDetail';
import ConfirmDeleteButton from '../util/ConfirmDeleteButton';
import { deleteMeal } from '../../actions/mealActions';
import '../../static/hover_overlay.scss';

class ConnectedMeal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      meal: props.meal,
    };
    this.handleOpenModal = this.handleOpenModal.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
    this.handleUpdate = this.handleUpdate.bind(this);
    this.handleDelete = this.handleDelete.bind(this);
  }

  handleOpenModal() {
    this.setState({ isOpen: true });
  }

  handleCloseModal() {
    this.setState({ isOpen: false });
  }

  handleDelete() {
    this.props.removeMeal(this.props.meal.id);
  }

  handleUpdate() {
    this.props.handleMealUpdate(this.props.meal);
  }

  render() {
    const { isDeleting, isUpdating } = this.props;

    const btnStyle = {
      flex: '50%',
      border: 'none',
      margin: '.5rem',
      padding: '.5rem',
      cursor: 'pointer',
      borderRadius: '3px',
      background: '#e9ebeb',
    };

    const modalStyle = {
      overlay: {
        backgroundColor: 'rgba(0, 0, 0, 0.25)',
      },
      content: {
        top: '50%',
        left: '50%',
        right: 'auto',
        bottom: 'auto',
        marginRight: '-50%',
        transform: 'translate(-50%, -50%)',
        minWidth: '20rem',
        width: '70%',
      },
    };

    return (
      <div className="card user-menu-item">
        <div
          className="overlay-container"
          onClick={this.handleOpenModal}
          onKeyPress={this.handleOpenModal}
          role="button"
          tabIndex="0"
        >
          <img
            src={this.state.meal.imageUrl}
            alt={this.state.meal.name}
          />
          <div className="overlay">
            <div className="text">
              <p>View description</p>
            </div>
          </div>
        </div>
        <Modal
          isOpen={this.state.isOpen}
          contentLabel="Meal Detail"
          style={modalStyle}
        >
          <Button
            value="&times;"
            onClick={this.handleCloseModal}
            className="close"
          />
          <MealDetail
            closeMealDetail={this.handleCloseModal}
            meal={this.state.meal}
          />
        </Modal>
        <div
          className="font-weight-bold text-black"
          style={{ padding: '.875rem' }}
        >
          <p>{this.state.meal.name}</p>
          <p style={{ fontSize: '1.25rem' }}>
            &#x20a6;{this.state.meal.price}
          </p>
        </div>
        <div style={{ display: 'flex', width: '75%', margin: '.5rem auto' }}>
          <button
            value="Edit"
            disabled={isUpdating}
            onClick={this.handleUpdate}
            style={Object.assign({}, btnStyle, { color: 'green' })}
            title="Edit Meal"
          >
            Edit
          </button>
          <ConfirmDeleteButton
            value="Delete"
            disabled={isDeleting}
            handleOk={this.handleDelete}
            style={Object.assign({}, btnStyle, { color: 'red' })}
            title="Delete Meal"
            resource="meal"
          >
            Delete
          </ConfirmDeleteButton>
        </div>
      </div>
    );
  }
}

ConnectedMeal.propTypes = {
  meal: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ])).isRequired,
  isDeleting: PropTypes.bool.isRequired,
  isUpdating: PropTypes.bool.isRequired,
  handleMealUpdate: PropTypes.func.isRequired,
  removeMeal: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  isDeleting: state.meals.isDeleting,
  isUpdating: state.meals.isUpdating,
});

const mapDispatchToProps = dispatch => ({
  removeMeal: id => dispatch(deleteMeal(id)),
});

const Meal = connect(mapStateToProps, mapDispatchToProps)(ConnectedMeal);

export default Meal;
