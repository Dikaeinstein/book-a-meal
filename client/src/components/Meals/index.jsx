import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import CatererMeals from './CatererMeals';
import AddMeal from './AddMeal';
import UpdateMeal from './UpdateMeal';
import Footer from '../util/Footer';
import Button from '../util/Button';
import errorHandler from '../util/errorHandler';

Modal.setAppElement('#root');

class ConnectedMeals extends Component {
  constructor() {
    super();
    this.state = {
      isOpen: false,
      updating: false,
      meal: {},
    };
    this.handleOpenModal = this.handleOpenModal.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
    this.handleMealUpdate = this.handleMealUpdate.bind(this);
    this.handleAddMeal = this.handleAddMeal.bind(this);
  }

  handleOpenModal() {
    this.setState({ isOpen: true });
  }

  handleCloseModal() {
    this.setState({ isOpen: false });
  }

  handleMealUpdate(meal) {
    this.setState({ updating: true, meal });
    this.handleOpenModal();
  }

  handleAddMeal() {
    this.setState({ updating: false });
    this.handleOpenModal();
  }

  render() {
    const modalStyle = {
      overlay: {
        backgroundColor: 'rgba(0, 0, 0, 0.25)',
      },
      content: {
        top: '50%',
        left: '50%',
        right: 'auto',
        bottom: 'auto',
        marginRight: '-50%',
        transform: 'translate(-50%, -50%)',
        minWidth: '20rem',
        width: '60%',
      },
    };
    const CatererMealsWithErrorHandling =
      errorHandler(CatererMeals, 'Error fetching meals');
    return (
      <div>
        <main
          className="bg-light main"
          style={{ minHeight: 'calc(100vh - 151px)' }}
        >
          <Button
            value="Add meal"
            className="btn btn-default"
            onClick={this.handleAddMeal}
            style={{ margin: '1rem 1rem 1rem 0' }}
          />
          <Modal
            isOpen={this.state.isOpen}
            contentLabel="Meal"
            style={modalStyle}
            closeTimeoutMS={150}
          >
            <Button
              value="&times;"
              onClick={this.handleCloseModal}
              className="close"
            />
            {!this.state.updating
              ? <AddMeal closeModal={this.handleCloseModal} />
              : <UpdateMeal
                meal={this.state.meal}
                closeModal={this.handleCloseModal}
              />}
          </Modal>
          <CatererMealsWithErrorHandling
            handleMealUpdate={this.handleMealUpdate}
            error={this.props.error}
          />
        </main>
        <Footer />
      </div>
    );
  }
}

ConnectedMeals.propTypes = {
  /* eslint react/require-default-props: 0 */
  error: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.objectOf(PropTypes.string),
  ]),
};

const mapStateToProps = state => ({
  error: state.meals.fetchError,
});

const Meals = connect(mapStateToProps)(ConnectedMeals);

export default Meals;
