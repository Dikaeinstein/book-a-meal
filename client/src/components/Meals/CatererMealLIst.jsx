import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Meal from './Meal';

const ConnectedCatererMealList = ({ meals, handleMealUpdate }) => {
  const mealList = meals.map(meal => (
    <Meal key={meal.id} meal={meal} handleMealUpdate={handleMealUpdate} />
  ));

  return (
    <section className="card user-menu text-left">
      {mealList}
    </section>
  );
};

ConnectedCatererMealList.propType = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleMealUpdate: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  meals: state.meals.data,
});

const CatererMealList = connect(mapStateToProps)(ConnectedCatererMealList);

export default CatererMealList;
