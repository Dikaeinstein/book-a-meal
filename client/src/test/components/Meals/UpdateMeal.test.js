import React from 'react';
import { Formik } from 'formik';
import { shallow } from 'enzyme';
import { UpdateMeal } from '../../../components/Meals/UpdateMeal';

describe('<UpdateMeal />', () => {
  const meal = {
    name: 'Test meal',
    description: 'This is a test meal',
    imageUrl: 'http://testimageurl',
    price: 4000,
  };
  const props = {
    editMeal: jest.fn(),
    closeModal: jest.fn(),
    meal,
  };
  it('should call handleSubmit when form is submitted', () => {
    const wrapper = shallow(<UpdateMeal {...props} />);
    const form = wrapper.find(Formik);
    form.simulate('submit');
    expect(props.editMeal).toHaveBeenCalled();
  });
  it('should match snapshot', () => {
    const tree = shallow(<UpdateMeal {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
