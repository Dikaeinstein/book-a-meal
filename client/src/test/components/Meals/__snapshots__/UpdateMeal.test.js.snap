// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateMeal /> should match snapshot 1`] = `
<section
  className="add-section"
>
  <h2
    className="text-center"
  >
    Update Meal
  </h2>
  <Formik
    component={[Function]}
    enableReinitialize={false}
    initialValues={
      Object {
        "description": "This is a test meal",
        "imageUrl": "http://testimageurl",
        "name": "Test meal",
        "price": 4000,
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": false,
        "_nodes": Array [
          "imageUrl",
          "price",
          "description",
          "name",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "description": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "imageUrl": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "undefined": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "name": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "price": NumberSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "number",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  />
</section>
`;
